modules/c++/math.poly/include/math/poly/Fit.h

```diff
@@ -114,30 +114,63 @@ inline math::poly::TwoD<double> fit(const math::linear::Matrix2D<double>& x,
                     size_t nx,
                     size_t ny)
 {
-    // Normalize the values in the matrix
     size_t m = x.rows();
     size_t n = x.cols();
-    
+    size_t mxn = x.size();
+
     if (m != y.rows())
         throw except::Exception(Ctxt("Matrices must be equally sized"));
 
     if (n != y.cols())
         throw except::Exception(Ctxt("Matrices must be equally sized"));
-    
+
+    // Compute min and max values for centering
```

Comment:
Since x and y are the same, this code can be cut in half if you put in a helper function that takes in the original matrix and provides back the centered matrix.  While you're at it, you could move the normalization and scalar division (which were already in the code and are also duplicated for x and y) into that function too.